docs_src/threat-models/secret-store/threat_model.md

```diff
@@ -0,0 +1,353 @@
+
+Primary assets are the assets at the level of the conceptual data model of the system and primarily represent "real-world" things.
+
+| AssetId | Name                | Description                         | Attack Points                  |
+| ------- | ------------------- | ----------------------------------- | ------------------------------ |
+| P-1     | Application secrets | The things we are trying to protect | In use, in transit, in storage |
+
+### Secondary Assets
+
+Secondary assets are assets are used to support or protect the primary assets and are usually implementation details versus being part of the conceptual data model.
+
+
+| AssetId | Name                      | Description                                                  | Attack Points                       |
+| ------- | ------------------------- | ------------------------------------------------------------ | ----------------------------------- |
+| S-1     | Vault service token       | Vault service tokens are issued per-service and used by services to authenticate to vault and retrieve per-service application secrets. | In-flight via API, at rest          |
+| S-3     | Vault token-issuing-token | Used by the token issuing service to create vault service tokens for other services. (Called out separately from S-1 due to its high privilege.) | In-flight via API, at rest          |
+| S-4     | Vault root token          | A special token created at Vault initialization time that has all capabilities and never expires. | In-flight via API, at rest          |
+| S-5     | Vault master key          | A root secret that encrypts all of Vault's other secrets.    | In-flight via API, at rest, in-use. |
+| S-6     | Vault data store          | A data store encrypted with the Vault master key that contains the contents of the vault. | In storage                          |
+| S-7     | Consul data store         | Back-end storage engine for vault data store.                | In storage                          |
+| S-8     | CA key                    | Private keys for on-device PKI certificate authority.        | In use, in transit, in storage      |
+| S-9     | Issuing CA key            | Private keys for on-device PKI issuing authorities.          | In use, in transit, in storage      |
+| S-10    | Leaf TLS key              | Private keys for TLS server authentication for on-device services (e.g. Vault service, Consul service) | In use, in transit, in storage      |
+| S-13    | IKM                       | Initial keying material as input to HMAC KDF                 | In use, in transit, in storage      |
+
+Note that asset S-9 (issuing CA key) is not currently implemented:
+in all current EdgeX releases all TLS leaf certificates are derived from the root CA.
+
+## Attack Surfaces
+
+This table lists components in the system architecture that have assets of potential value to an attacker and how a potential attacker may attempt to gain access to those components.
```

Comment:
What about the devices/sensors and potential DoS attacks via those?  Imagine a sensor that starts spewing 1000s of readings a second vs one a second

I think we can put these out of scope for the threat model, but shouldn't it be mentioned?

Reply:
This threat model is scoped to the secret management subsystem only.